api.go

```
@@ -941,6 +951,99 @@ func (api *API) GetLogOutput(ctx context.Context, logType string, getLogRequest
 	return &models.GetLogOutputResponse{Log: string(logData)}, nil
 }
 
+func (api *API) BasicBackup(basicBackupRequest *models.BasicBackupRequest, w io.Writer) error {
+	var err error
+
+	lnStorageDir := ""
+
+	if api.svc.lnClient != nil {
+		lnStorageDir, err = api.svc.lnClient.GetStorageDir()
+		if err != nil {
+			return fmt.Errorf("failed to get storage dir: %w", err)
+		}
+
+		// Stop the client to ensure its storage is in a consistent state.
+		err = api.svc.StopLNClient()
+		if err != nil {
+			return fmt.Errorf("failed to stop LNClient: %w", err)
+		}
+	}
+
+	// Run online backup of the database.
+	now := time.Now()
+	nwcBkpFile := fmt.Sprintf("nwc-backup-%s.db", now.Format("060102150405"))
```

Why is this done? I thought we would just disconnect from the database and then copy the existing db file into the zip. We also don't want to change the name because it needs to be imported as the same filename

---

I was trying to make the backup mechanism as flexible as possible. So I decided not to shut down the database connection, but to run an online backup instead: this way, the app is not left in an inconsistent state, and can easily re-connect and proceed to run when the backup is complete.

The filename here is temporary; the database is always archived as *nwc.db*. I was trying to anticipate a scenario when the target NWC instance is configured to use a different database name. In this case, the restore procedure could rename the database file as appropriate.

---

That makes sense, thanks for the explanation! I guess my implementation supports this flow — we only need to make sure we shut down the app after the HTTP response is sent.

---

That would be great! So then you should be able to use the same logic as you do with the LDK files, right?

---

Updated the PR to shut down the app services (Nostr, LNClient) and close the database. TBH I still feel the online backup option is a more reliable way to do it, since it avoids the weird state of the app 🤔 

But anyways, how do we ensure that the configs on the old and the new machine match? I.e. if the old machine has a custom DB path, do we require the user to use the same on the new one?

---

@rdmitr maybe first we extract everything to the new app's WORK_DIR, that should be the default and work with most setups.

Could we throw an exception when importing for now for the other cases? (if the DATABASE_URI starts with `file:` or is an absolute or relative path which does not resolve to `$WORK_DIR/nwc.db`)

---

Right, I missed that. But that's exactly the case that was covered by the online backup ;)

---

One more question. Earlier in this thread you mentioned:

> We also don't want to change the name because it needs to be imported as the same filename

So, should we import it with the same filename, or use nwc.db always? I'm leaning towards the latter, as there's more certainty.